rust/lance-encoding/src/encoder.rs

```diff
@@ -810,6 +810,19 @@ impl CompressionStrategy for CoreArrayEncodingStrategy {
         }
         if let DataBlock::VariableWidth(ref variable_width_data) = data {
             if variable_width_data.bits_per_offset == 32 {
+                let data_size = variable_width_data.get_stat(Stat::DataSize).expect(
+                    "VariableWidth DataBlock should have valid `Stat::DataSize` statistics",
+                );
+                let data_size = data_size.as_primitive::<UInt64Type>().value(0);
+
+                let max_len = variable_width_data.get_stat(Stat::MaxLength).expect(
+                    "VariableWidth DataBlock should have valid `Stat::DataSize` statistics",
+                );
+                let max_len = max_len.as_primitive::<UInt64Type>().value(0);
+
+                if max_len > 4 && data_size >= 4 * 1024 * 1024 {
```

Comment:
I will do more experiments to tune this

Reply:
Sounds good.  The original 4MIB threshold was very arbitrary.